source/fundamentals/linq.txt

```diff
@@ -248,6 +248,44 @@ The result of the preceding example contains the following documents:
     { "name" : "Crystal Room", "cuisine" : "Italian" }
     { "name" : "Forlinis Restaurant", "cuisine" : "Italian" }
 
+.. _csharp-linq-takewhile:
+
+Limit on a Condition
+````````````````````
+
+You can use the ``TakeWhile()`` LINQ method in a ``Select()`` projection
+to return array field elements while a specified condition is true, then
+skip the remaining elements.
+
+This example uses the following ``Student`` class to model documents
+that contain an array field ``Grades``:
+
+.. code-block:: csharp
+
+   public class Student
+   {
+       public ObjectId Id { get; set; }
+       public string Name { get; set; }
+       public int[] Grades { get; set; }
+   }
+
+The following code shows how to use the ``TakeWhile()`` method to return
+any ``Grades`` array elements that are greater than ``90`` and skip the
+rest of the array:
+
+.. code-block:: csharp
+
+   var query = queryableCollection
+       .Select(s => s.Grades.TakeWhile(g => g > 90).ToArray());
+
+The results might resemble the following documents:
```

Comment:
The results look right, but calling them "documents" is not quite right.

Maybe just say "The results might resemble the following:"